### ISSUE TITLE
JavaScript|Intermediate Git| A Deeper Look at Git: Assignments are too high level for what is discussed/needed at that point

### ISSUE THREAD

**Issue by emergent-code** [TheOdinProject/curriculum#28426]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

Currently, the assignments in this section are:

1. Read the chapter on [Rebasing covered by git-scm](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) for an even deeper dive into Rebasing.
2. Read the chapter on [Reset covered by git-scm](https://git-scm.com/book/en/v2/Git-Tools-Reset-Demystified) for a deeper dive into git reset.

Git-scm, while a good and comprehensive resource can make the topic more confusing to learn for a beginner. Also, a lot of what is covered seems to be unnecessary at this stage. Having to read all of this creates a lot of cognitive load, especially when it is not pointed out what is essential and and what isn't. There is also no element of practice after all of that reading.

I propose a gentler introduction to intermediate git as shown below:

**### Assignments**

1. Watch [How Git Works: Explained in 4 minutes](https://www.youtube.com/watch?v=e9lnsKot_SQ&pp=ygUDZ2l0)
2. Read [think-like-a-git](http://think-like-a-git.net/) (moved from the following section to this section)
3. Complete the first three sub-sections in the main section of [Learn Git Branching](https://learngitbranching.js.org/)*
4. Watch [Git MERGE vs REBASE: Everything You Need to Know](https://www.youtube.com/watch?v=0chZFIZLR_0&pp=ygUGcmViYXNl)

*This resource is especially good for building fluency/familiarity with the commands and for being able to visualise what is happening. I'd also recommend the remote section as an assignment for the next lesson.

I think my suggested changes result in a set of assignments which are shorter and provide more clarity.

I'd add the git-scm articles as additional resources.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-a-deeper-look-at-git

**Comment by wise-king-sullyman** [comment]
Thanks for taking the time to make this issue!

So, I think you raise some interesting points here, but one of the difficult things about this content is that there really isn't any specific point in the curriculum where this deeper level Git stuff will be truly "necessary" to complete the further lessons and projects.

We dive so deep into it despite that because it's great knowledge to have when you're trying to get/in your first dev role that _really_ helps a jr stand out. Additionally git-scm is *the* Git resource, and while I can understand the content being a bit of a slog getting more comfortable with highly technical documentation like it is one of the big skills we're trying to build here.

All of that being said, I think I would be hesitant to pull those assignments out, but there might be a solid case for tweaking the positioning of the content in the curriculum a bit more. I will mull your proposal over further though.

CC @TheOdinProject/maintainers for comments from other members of the maintainer team.

**Comment by emergent-code** [comment]
Thanks for the consideration. I'd say maybe just add parts of https://learngitbranching.js.org/ then. 

Part of the reason why the git-scm resource feels like a lot is because you read all of it, but then you don't practise it. There's no "tangibility" to what you've learned for lack of a better word and any assumed expertise would fall apart on closer examination. It's like assigning a chapter of a university level calculus intro textbook for reading to a high school class without giving them the opportunity to practice. It probably does help a junior dev stand out, but how much time will elapse between someone reading the resources and interviewing for a job, and how much will they remember? I'd argue having really solid and trainable fundamentals is better than being aware of something high level that you have no idea how to use in practice. By the time it comes to actually using the things mentioned on git-scm it would be a breeze to learn. Anyway, that's just my 2 cents on it, I hope it doesn't come across in a negative tone.

**Comment by wise-king-sullyman** [comment]
> Anyway, that's just my 2 cents on it, I hope it doesn't come across in a negative tone.

Definitely not, I appreciate you taking the time to give your thoughts!

I wouldn't be opposed to some of learngitbranching being moved to an assignment instead of all just being an additional resource.

**Comment by wise-king-sullyman** [comment]
@emergent-code looks like no other maintainers have voiced opposition, is this still something you'd be interested in doing?

**Comment by emergent-code** [comment]
> @emergent-code looks like no other maintainers have voiced opposition, is this still something you'd be interested in doing?

Yeah, of course. So just to be clear, I'd be adding some of  https://learngitbranching.js.org/ to assignments?

**Comment by wise-king-sullyman** [comment]
Sorry missed your reply, yes that's what this issue would entail.

**Comment by KevinMulhern** [comment]
Hey @emergent-code, are you still working on this? or would you like us to give someone else the opportunity to do it?

**Comment by emergent-code** [comment]
Ah, completely forgot about this. I should have some time in the coming week or so to do it, but if it's urgent, you can give the opportunity to someone else!

**Comment by KevinMulhern** [comment]
Thanks @emergent-code - it's not urgent

**Comment by wise-king-sullyman** [comment]
Hey @emergent-code, just touching base, is this something you still plan to work on?

**Comment by carlosmperezm** [comment]
Hi 👋🏻, just to make sure I understood correctly, the first three sub-sections in the main section of [Learn Git Branching](https://learngitbranching.js.org/) are the only things that need to be added, right?

The assignment will look something like: 
> [Learn Git Branching](https://learngitbranching.js.org/) is an interactive game on the web to learn git. Go ahead and do the first three sub-sections under the **Introduction Sequence**. Those are:
> 1. Introduction to Git Commits
> 2. Branching in Git
> 3. Merging in Git

?

If that's the case, I can work on this if you'd like.

**Comment by mao-sz** [comment]
@carlosmperezm Yep, but I'd recommend just assigning the first section "Introduction Sequence" as a whole rather than only a few subsections. Add that to the "A Deeper Look at Git" lesson assignment (probably as a first item would be best), then remove the "Learn Git Branching" additional resource from "Working with Remotes".
